### PR TITLE
cache: remove buffer_acquire/release part 3

### PR DESCRIPTION
**Pull request by tobonex**
This is step 5 of implementation of #8006
This PR is a result of splitting https://github.com/thesofproject/sof/pull/8106 into parts.

Remove usage of buffer_acquire and buffer_release from a number of files.

next part:https://github.com/thesofproject/sof/pull/8219

### CODE REVIEW

**Review by btian1 — APPROVED** [review]
could you change commit message to
this is follow up with changes:4a03669

File: src/audio/kpb.c [hunk 1]
```diff
@@ -1729,7 +1707,6 @@ static void kpb_init_draining(struct comp_dev *dev, struct kpb_client *cli)
 static enum task_state kpb_draining_task(void *arg)
 {
 	struct draining_data *draining_data = (struct draining_data *)arg;
-	struct comp_buffer *sink = buffer_acquire(draining_data->sink);
```
Comment (lyakh) [inline src/audio/kpb.c:1732]: I'd keep the pointer - it's used often enough and it would reduce the diff

**Comment by tobonex** [comment]
> could you change commit message to this is follow up with changes:4a03669

Changing all of the commits is a bit of work so let's clear it up a little. You want me to change `this is a continuation of changes
from commit 4a03669` to `this is follow up with changes:4a03669` or the whole commit message to `this is follow up with changes:4a03669`?

**Review by marcinszkudlinski — APPROVED** [review]

File: src/audio/kpb.c (hunk 1 above)
Comment (tobonex) [reply src/audio/kpb.c:1753]: Ok, kept it.

**Review by kv2019i — APPROVED** [review]

**Comment by kv2019i** [comment]
https://github.com/thesofproject/sof/issues/8238 causing fails in Intel pr-device-test , rest is good. We have a problem with the github flow for Windows builds -- the ninja is lost, now all the time. FYI @marc-hb  @marcinszkudlinski do we have any "volunteer" to debug Windows builds now that @aborisovich can no longer help with these.

**Pull request merged**